app/src/context/evm/prompt-wars-market-contract/PromptWarsMarketContractContextController.tsx

```diff
@@ -0,0 +1,377 @@
+    return PromptWarsMarketContractStatus.RESOLVED;
+  }
+
+  if (values?.isOver && !values.isRevealWindowExpired) {
+    return PromptWarsMarketContractStatus.REVEALING;
+  }
+
+  if (values?.isOver && !values.isResolutionWindowExpired) {
+    return PromptWarsMarketContractStatus.RESOLVING;
+  }
+
+  if (values?.isOver && values.isExpiredUnresolved) {
+    return PromptWarsMarketContractStatus.UNRESOLVED;
+  }
+
+  return PromptWarsMarketContractStatus.CLOSED;
+};
+
+let marketContract: Market;
+
+export const PromptWarsMarketContractContextController = ({
+  marketId,
+  children,
+}: PromptWarsMarketContractContextControllerProps) => {
+  const routes = useRoutes();
+  const router = useRouter();
+  const toast = useToastContext();
+  const signer = useEthersSigner();
+
+  const [marketContractValues, setMarketContractValues] = useState<PromptWarsMarketContractValues>();
+
+  const [actions, setActions] = useState<PromptWarsMarketContractContextContextActions>({
+    fetchMarketContractValues: {
+      isLoading: false,
+    },
+    ftTransferCall: {
+      success: false,
+      isLoading: false,
+    },
+    create: {
+      isLoading: false,
+    },
+  });
+
+  const { isConnected } = useAccount();
+
+  const assertWalletConnection = () => {
+    if (!isConnected) {
+      toast.trigger({
+        variant: "info",
+        withTimeout: true,
+        title: "Wallet is not connected",
+        children: <Typography.Text>Check your internet connection, your balance and try again.</Typography.Text>,
+      });
+
+      throw new Error("ERR_PROMPT_MARKET_CONTRACT_INVALID_WALLET_CONNECTION");
+    }
+  };
+
+  const fetchMarketContractValues = async () => {
+    setActions((prev) => ({
+      ...prev,
+      fetchMarketContractValues: {
+        isLoading: true,
+      },
+    }));
+
+    try {
+      // Wait 1 second to allow flags to change
+      setTimeout(async () => {
+        try {
+          if (!signer) throw new Error("not signed in");
+
+          const contract = await connect(DEPLOYED_CONTRACT_ADDRESS, signer);
+
+          const [
+            market,
+            resolution,
+            fees,
+            management,
+            collateralToken,
+            // outcomeIds,
+            // isResolved,
+            // isOpen,
+            // isOver,
+            // isRevealWindowExpired,
+            // isResolutionWindowExpired,
+            // isExpiredUnresolved,
+          ] = await Promise.all([
+            contract.get_market_data(),
+            contract.get_resolution_data(),
+            contract.get_fees_data(),
+            contract.get_management_data(),
+            contract.get_collateral_token_data(),
+            // contract.get_outcome_ids(),
+            // contract.is_resolved(),
+            // contract.is_open(),
+            // contract.is_over(),
+            // contract.is_reveal_window_expired(),
+            // contract.is_resolution_window_expired(),
+            // contract.is_expired_unresolved(),
```

Comment:
Currently in the solidity contract 
1) get_outcome_ids is not present
2) is_resolved is private _is_resolved 
3) No is_open
4) No is_over
5) _is_reveal_window_expired is private
6) _is_resolution_window_expired is private
7) _is_expired_unresolved is private

Reply:
Yeah, there was some refactor towards simplicity and efficiency in the new Solidity game. We should be able to play the game with the methods the contract has. 

Try looking at the contract tests, all the methods are covered, including the full game cycle: https://github.com/Udit-takkar/pulsemarkets/blob/feat/typescript-interface/solidity/promptwars/test/index.ts

Regarding the private methods, we can make them public indeed. Let me work on that for you.